pkgs/development/haskell-modules/with-packages-wrapper.nix

```diff
@@ -46,14 +47,18 @@ let
   llvm          = lib.makeBinPath
                   ([ llvmPackages.llvm ]
                    ++ lib.optional stdenv.isDarwin llvmPackages.clang);
+  ghcLibdirLink = runCommand "ghc_libdir" {} ''
+    mkdir -p '${libDir}'
+    echo '${ghcLibdir}' > '${libDir}/ghc_libdir_override'
```

Comment:
Actually, I think I know why this is used. Last time I checked, `buildEnv` would try to create as few directories as possible, instead attempting to symlink as many existing directories as possible. If two entries in `paths` have the same directory, it would then be forced to create a new directory and symlink the *contents* of each of those two entries into that new directory.

Because there's an option (in this patch) to allow setting a default setting for `ghc_libdir` in the `ghcjs/base.nix` expression, we need to toss in something like `ghcLibdirLink` in `paths` to force `baseEnv`'s hand, so the user can supply an overriding setting in `ghcWithPackages`. Gross, but it works.

`pkgs.symlinkJoin` behaves much better here, as it always creates directories in the derivation in outputs, rather than trying to use symlinks everywhere.

@peti is there any downside you can see in switching from `buildEnv` to `symlinkJoin` here?